PostHog/PostHogRemoteConfig.swift

```diff
@@ -450,6 +461,16 @@ class PostHogRemoteConfig {
         return value
     }
 
+    // To be called after acquiring `requestIdLock`
```

Comment:
can we actually reuse `featureFlagsLock` instead? `requestId` is only set/cleared within the flags context anyway